### PR TITLE
Lots of README changes

### PR DESCRIPTION
**Pull request by rockerBOO**
- Add back prior intro line
- Add What is Telescope? 
_(saw multiple people read the intro line, then it has a image, then there is a block of text about what it does. This directs it before the image.)_
- Add Theme image to encourage people to play with their layouts more. 
- Added all the sections to the ToC and renamed some of the built in redundancies 
_(We can be more specific of external commands instead of saying everything is builtin.)_

### CODE REVIEW

**Review by tjdevries — APPROVED** [review]
LGTM. Tami can make any comments required.

File: README.md [hunk 1]
```diff
@@ -1,36 +1,31 @@
...
 
-`Telescope.nvim` is a next generation library for creating floating pickers
-with advanced features.  It is written in `lua` and is built on top of latest
-awesome features from `neovim` core.  `Telescope.nvim` is centered around
-modularity so much so that each picker is its own world, (meaning it can be
-configured in isolation from other pickers, such presentation, algorithm,
-mappings ... etc).
+`telescope.nvim` is a highly extendable fuzzy finder over lists.   Built on top of latest
+awesome features from `neovim` core.  Telescope is centered around
+modularity, allowing for easy customization.
```
Comment (tami5) [inline README.md:10]: Here (after around modularity) perhaps we need to mention to what extent telescope is modular. Modularity is very general feature that could mean different things.

File: README.md [hunk 2]
```diff
@@ -1,36 +1,31 @@
...
 
-`Telescope.nvim` was built with the intention of becoming a library, but it has long grown to
-be much bigger than that.  In addition to
-[extensions](https://github.com/nvim-telescope) `Telescope.nvim` comes with a
-growing number of community driven [built-in pickers](#built-in-pickers),
-covering a wide range of use cases and tools, and offers a customizable user
-interface.
+![by @glepnir](https://user-images.githubusercontent.com/41671631/100819597-6f737900-3487-11eb-8621-37ec1ffabe4b.gif)
 
+`telescope.nvim` has a number of [extensions](https://github.com/nvim-telescope) for supporting specific plugins.
```
Comment (tami5) [inline README.md:14]: Builtin picker should be mentioned before the external extensions.

**Review by tami5 — CHANGES_REQUESTED** [review]
LGMT @rockerBOO, I agree the introduction was a bit bulky

File: README.md (hunk 1 above)
Comment (rockerBOO) [reply README.md:10]: I think its all modular so everything can be swapped out? From the end user, this should be accurate.

File: README.md (hunk 2 above)
Comment (rockerBOO) [reply README.md:14]: I get that added!

File: README.md (hunk 1 above)
Comment (rockerBOO) [reply README.md:10]: Maybe we say the sorters, pickers, finders can all be composed modularly?

Comment (tami5) [reply README.md:10]: Oh thats it. that should do it

**Comment by rockerBOO** [comment]
@tami5 we can refine that phrase in another pull if you have any suggestions.

**Pull request merged**